### DISCUSSION TITLE
teslmate Stopped gathering data since a month !

### DISCUSSION THREAD

**Discussion by leelouch** [teslamate-org/teslamate#3562] [Q&A] [answered]
Hi, I was running teslamate 1.27.2 for almost one year now ! everything was working well, 
I noticed today that teslamate stopped gathering data ! 
I upgraded the docker to the latest version, but still have the same problem. 
I am running teslamate on synology docker 
here is some logs and screenshot
Thanks for your help ! 

Teslamate Version : 1.28.2
Grafana  version : 10.1.2

I noticed an error in the log : 2023-12-12 18:10:35.129 car_id=1 [error] Error / not_signed_in
is this normal ? 
Thanks for your help 

```
2023-12-12 18:08:50.310 [info] Migrations already up
2023-12-12 18:08:54.207 [info] System Info: Erlang/OTP 26 (jit)
2023-12-12 18:08:54.207 [info] Version: 1.28.2
2023-12-12 18:08:54.615 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (330.836 ms)
2023-12-12 18:08:54.615 [info] Refreshed api tokens
2023-12-12 18:08:54.671 [info] Scheduling token refresh in 6 h
2023-12-12 18:08:54.676 [info] Running TeslaMateWeb.Endpoint with cowboy 2.10.0 at :::4000 (http)
2023-12-12 18:08:54.677 [info] Access TeslaMateWeb.Endpoint at http://192.168.0.100
2023-12-12 18:08:54.814 [info] Starting logger for 'Rocky'
2023-12-12 18:08:54.835 [info] MQTT connection has been established
2023-12-12 18:08:54.920 car_id=1 [info] Start / :asleep
2023-12-12 18:08:57.428 [info] GET /
2023-12-12 18:08:57.436 [info] Sent 200 in 7ms
2023-12-12 18:09:51.492 [info] GET /
2023-12-12 18:09:51.498 [info] Sent 200 in 6ms
2023-12-12 18:10:25.460 [info] GET /settings
2023-12-12 18:10:25.477 [info] Sent 200 in 17ms
2023-12-12 18:10:33.966 [info] GET /
2023-12-12 18:10:33.971 [info] Sent 302 in 4ms
2023-12-12 18:10:33.975 [info] GET /sign_in
2023-12-12 18:10:33.982 [info] Sent 200 in 6ms
2023-12-12 18:10:35.129 car_id=1 [error] Error / not_signed_in
2023-12-12 18:10:46.409 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (475.475 ms)
2023-12-12 18:10:46.496 [info] Scheduling token refresh in 6 h
2023-12-12 18:10:46.574 [info] Starting logger for 'Rocky'
2023-12-12 18:10:46.709 car_id=1 [info] Start / :asleep
2023-12-12 18:10:46.762 [info] GET /
2023-12-12 18:10:46.770 [info] Sent 200 in 7ms
2023-12-12 18:11:11.356 [info] GET /
2023-12-12 18:11:11.361 [info] Sent 200 in 5ms
2023-12-12 18:11:15.300 [info] GET /settings
2023-12-12 18:11:15.314 [info] Sent 200 in 13ms
2023-12-12 18:11:57.041 [info] GET /
2023-12-12 18:11:57.045 [info] Sent 302 in 3ms
2023-12-12 18:11:57.049 [info] GET /sign_in
2023-12-12 18:11:57.056 [info] Sent 200 in 6ms
2023-12-12 18:12:00.236 [info] GET /
2023-12-12 18:12:00.241 [info] Sent 302 in 4ms
2023-12-12 18:12:00.245 [info] GET /sign_in
2023-12-12 18:12:00.252 [info] Sent 200 in 6ms
2023-12-12 18:12:13.761 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (262.318 ms)
2023-12-12 18:12:13.814 [info] Scheduling token refresh in 6 h
2023-12-12 18:12:13.852 [info] Starting logger for 'Rocky'
2023-12-12 18:12:13.927 car_id=1 [info] Start / :asleep
2023-12-12 18:12:14.074 [info] GET /
2023-12-12 18:12:14.079 [info] Sent 200 in 5ms
```

![image](https://github.com/teslamate-org/teslamate/assets/9366148/ed641741-29f0-4ee6-9810-2353607f9b03)

![image](https://github.com/teslamate-org/teslamate/assets/9366148/731109a1-34c0-4c13-9e00-5ed8a7158573)

**Comment by cwanja** [comment, marked as answer]
What Tesla car software version are running / were running? 

Wake your car up and see if data starts logging. Per the logs, your TeslaMate instance is connected, has tokens and able to talk to your car.
```
2023-12-12 18:08:54.207 [info] Version: 1.28.2
2023-12-12 18:08:54.615 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (330.836 ms)
2023-12-12 18:08:54.615 [info] Refreshed api tokens
2023-12-12 18:08:54.671 [info] Scheduling token refresh in 6 h
```
**Note:** if it ends up that your TeslaMate instance was down or connection was broken, there is no way to backfill the data from Tesla.

**Comment by leelouch** [comment]
Hi, 
**Thanks for the quick answer !** 
My Car version actualy is 2023.44.1, 
It was before stopping gathering data:  2023.38.6

I woke up the car by openning the tesla up and indeed teslamate was able to gather data ! 
following the log and screenshot : 
```
2023-12-12 18:59:23.340 [info] GET /
2023-12-12 18:59:23.348 [info] Sent 200 in 7ms
2023-12-12 19:00:01.674 car_id=1 [info] Start / :online
2023-12-12 19:00:01.680 car_id=1 [info] Logged missing software update: 2023.44.1 d0b7099fb39d
2023-12-12 19:00:02.661 car_id=1 [info] Connecting ...
2023-12-12 19:03:05.052 car_id=1 [info] Suspending logging
2023-12-12 19:03:05.695 car_id=1 [warning] Received stale stream data: %TeslaApi.Stream.Data{time: ~U[2023-12-12 19:03:07.525Z], speed: nil, odometer: 11817.2, soc: 67, elevation: 45, est_heading: 94, est_lat: 48.865421, est_lng: 2.271636, power: 0, shift_state: nil, range: 190, est_range: 188, heading: 94}
```
![image](https://github.com/teslamate-org/teslamate/assets/9366148/13e99f21-810a-4912-a5cf-e37b51934c97)

**What does this mean now and why this happend,  is this a random behavior ? a bug ? will it work now ?**

**Reply by cwanja** [reply]
Lengthly detail read in #3393. Suggest to subscribe to GitHub releases and update accordingly. 

TLDR - there was a Tesla API change and it broke TeslaMate. Since https://github.com/teslamate-org/teslamate/releases/tag/v1.27.4, TeslaMate is working (re: suggestion to subscribe and update TeslaMate regularly). It will work for the foreseeable future, but Tesla is planning API changes and could break TeslaMate in the future. 

Long reads, from a separate comment:
> See #3371, #3370, #3389 and #3386. A "fix" is in #3370

**Comment by leelouch** [comment]
Thanks a lot for your quick help, I will go over those tickets and subscribe to github releases.

**Comment by passerby2024** [comment]
I already updated to the recent Teslamate build about two weeks ago.
But I'm having a similar issue since 2023.38 update.  

For example, everything seemed fine until I parked my car yesterday afternoon.  
Now it's in the same place, but the dashboard says "Health check failed" and the map is not shown properly.
I woke the car up to see what happens, but no change.

![image](https://github.com/teslamate-org/teslamate/assets/153710344/5101a949-8efa-47d9-a96f-056d81e19d6c)

One day I was charging my car, but Teslamate did not record the charging session at all and recorded "asleep".  
(although I could control my car wirelessly, which means it was online)
